client/src/main/kotlin/io/spine/examples/chatspn/desktop/ProfilePage.kt

```diff
@@ -0,0 +1,369 @@
+                model.deleteChat(chat.value!!.id)
+            },
+            chat.value!!
+        )
+        model.openModal(content)
+    }
+}
+
+/**
+ * Represents an email field.
+ */
+@Composable
+private fun EmailField(email: String) {
+    Box(Modifier.clip(MaterialTheme.shapes.small)) {
+        Column(
+            Modifier
+                .background(MaterialTheme.colorScheme.background)
+                .fillMaxWidth()
+                .padding(12.dp, 8.dp),
+            verticalArrangement = Arrangement.Center
+        ) {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Icon(
+                    Icons.Default.Email,
+                    "Email",
+                    Modifier.size(20.dp),
+                    MaterialTheme.colorScheme.primary
+                )
+                Spacer(Modifier.width(4.dp))
+                Text(
+                    "Email",
+                    color = MaterialTheme.colorScheme.primary,
+                    style = MaterialTheme.typography.headlineMedium
+                )
+            }
+            Spacer(Modifier.height(4.dp))
+            Text(
+                email,
+                style = MaterialTheme.typography.bodyLarge
+            )
+        }
+
+    }
+}
+
+/**
+ * Represents a default button on the info page.
+ */
+@Composable
+private fun InfoPageButton(
+    text: String = "",
+    icon: ImageVector? = null,
+    contentColor: Color = MaterialTheme.colorScheme.primary,
+    onClick: () -> Unit
+) {
+    Box(Modifier.clip(MaterialTheme.shapes.small)) {
+        Button(
+            onClick,
+            Modifier
+                .fillMaxWidth()
+                .pointerHoverIcon(PointerIcon(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR))),
+            shape = MaterialTheme.shapes.small,
+            colors = ButtonDefaults.buttonColors(
+                containerColor = MaterialTheme.colorScheme.background,
+                contentColor = contentColor
+            ),
+            contentPadding = PaddingValues(12.dp, 8.dp)
+        ) {
+            Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+                if (icon != null) {
+                    Icon(
+                        icon,
+                        text,
+                        Modifier.size(20.dp),
+                        contentColor
+                    )
+                    Spacer(Modifier.width(4.dp))
+                }
+                Text(
+                    text,
+                    color = contentColor,
+                    style = MaterialTheme.typography.bodyLarge
+                )
+            }
+        }
+    }
+}
+
+/**
+ * UI Model for the [ProfilePage].
+ */
+private class ProfilePageModel(
+    private val client: DesktopClient,
+    val userProfile: MutableState<UserProfile>,
+    val chatState: MutableState<ChatData?>,
+    val onBack: () -> Unit,
+    val openModal: (content: @Composable () -> Unit) -> Unit,
```

Comment:
It is not clear why we store the content of the modal window component in the model. Please explain this in the documentation.